target/linux/ramips/files/drivers/net/ethernet/ralink/gsw_mt7620.c

```diff
@@ -90,19 +90,15 @@ static void mt7620_hw_init(struct mt7620_gsw *gsw, int mdio_mode)
 	u32 val;
 	u32 is_BGA = (rt_sysc_r32(0x0c) >> 16) & 1;
 
-	rt_sysc_w32(rt_sysc_r32(SYSC_REG_CFG1) | BIT(8), SYSC_REG_CFG1);
```

Comment:
@blogic @nbd168 I am wondering why this line was included in the ethernet driver at all....any specific reason?

Reply:
Just checked the vendor driver and it has this in ethernet driver without any explanation:
```c
        /* PCIE_RC_MODE=1 */
        *(volatile u32 *)(REG_SYSCFG1) |= (0x1 << 8);
```
We should probably keep this vendor magic numbers when nobody knows what it actually does.

Reply:
Thanks for confirming that it is not a random mistake, but I still think it doesn't belong here.

According to the mt7620 programming guide, it sets PCIe to Root Complex mode, and the comment line from the vendor driver also confirms that.

The programming guide also states the initial value is set via Bootstrapping, and I have noticed no difference when removing this line.

Since both sources suggest it has nothing to do with ethernet, it seems safe to remove...or maybe it would be better to put this in the PCI driver where it belongs? probably in the function `mt7620_pci_hw_init`?

see page 22
http://download.villagetelco.org/hardware/MT7620/MT7620_ProgrammingGuide.pdf

Reply:
I found a good explanation in the programming guide

"FE/GE port is not available when operating in PCIe iNIC mode"

meaning that when PCIe is in Endpoint mode, ethernet gets disabled
so I will keep the line and add a comment